packages/@aws-cdk/aws-lambda-event-sources/test/test.kafka.ts

```diff
@@ -0,0 +1,264 @@
+        {
+          Type: 'VPC_SUBNET',
+          URI: {
+            Ref: 'VpcPrivateSubnet1Subnet536B997A',
+          },
+        },
+        {
+          Type: 'VPC_SUBNET',
+          URI: {
+            Ref: 'VpcPrivateSubnet2Subnet3788AAA1',
+          },
+        },
+      ],
+    }));
+
+    test.done();
+  },
+
+  'self managed Kafka in VPC: subnet and securityGroup must be set together'(test: Test) {
+    const stack = new cdk.Stack();
+    const fn = new TestFunction(stack, 'Fn');
+    const kafkaTopic = 'some-topic';
+    const secret = new Secret(stack, 'Secret', { secretName: 'AmazonMSK_KafkaSecret' });
+    const bootstrapServers = ['kafka-broker:9092'];
+    const vpc = new Vpc(stack, 'Vpc');
+
+    test.throws(() => {
+      fn.addEventSource(new sources.SelfManagedKafkaEventSource(
+        {
+          bootstrapServers: bootstrapServers,
+          topic: kafkaTopic,
+          secret: secret,
+          startingPosition: lambda.StartingPosition.TRIM_HORIZON,
+          vpc: vpc,
+          vpcSubnets: { subnetType: SubnetType.PRIVATE },
+        }));
+    }, /both subnets and securityGroup must be set/);
+
+    test.done();
+  },
+
+  'self managed Kafka: using SCRAM-SHA-256'(test: Test) {
+    // GIVEN
+    const stack = new cdk.Stack();
+    const fn = new TestFunction(stack, 'Fn');
+    const kafkaTopic = 'some-topic';
+    const secret = new Secret(stack, 'Secret', { secretName: 'AmazonMSK_KafkaSecret' });
+    const bootstrapServers = ['kafka-broker:9092'];
+    const sg = SecurityGroup.fromSecurityGroupId(stack, 'SecurityGroup', 'sg-0123456789');
+    const vpc = new Vpc(stack, 'Vpc');
+
+    // WHEN
+    fn.addEventSource(new sources.SelfManagedKafkaEventSource(
+      {
+        bootstrapServers: bootstrapServers,
+        topic: kafkaTopic,
+        secret: secret,
+        startingPosition: lambda.StartingPosition.TRIM_HORIZON,
+        vpc: vpc,
+        vpcSubnets: { subnetType: SubnetType.PRIVATE },
+        securityGroup: sg,
+        authenticationMethod: 'SASL_SCRAM_256_AUTH',
+      }));
+
+    expect(stack).to(haveResource('AWS::Lambda::EventSourceMapping', {
+      FunctionName: {
+        Ref: 'Fn9270CBC0',
+      },
+      BatchSize: 100,
+      SelfManagedEventSource: {
+        Endpoints: {
+          KafkaBootstrapServers: bootstrapServers,
+        },
+      },
+      StartingPosition: 'TRIM_HORIZON',
+      Topics: [
+        kafkaTopic,
+      ],
+      SourceAccessConfigurations: [
+        {
+          Type: 'SASL_SCRAM_256_AUTH',
+          URI: {
+            Ref: 'SecretA720EF05',
+          },
+        },
+        {
+          Type: 'VPC_SECURITY_GROUP',
+          URI: 'sg-0123456789',
+        },
+        {
+          Type: 'VPC_SUBNET',
+          URI: {
+            Ref: 'VpcPrivateSubnet1Subnet536B997A',
+          },
+        },
+        {
+          Type: 'VPC_SUBNET',
+          URI: {
+            Ref: 'VpcPrivateSubnet2Subnet3788AAA1',
+          },
+        },
+      ],
+    }));
```

Comment:
Use the `arrayWith()` API from `@aws-cdk/assert` module to make this simpler -

```suggestion
    expect(stack).to(haveResourceLike('AWS::Lambda::EventSourceMapping', {
      SourceAccessConfigurations: arrayWith(
        {
          Type: 'SASL_SCRAM_256_AUTH',
          URI: {
            Ref: 'SecretA720EF05',
          },
        },
      ),
    }));
```